packages/react-components/source/react/internal/option-menu-list/OptionMenuList.js

```diff
@@ -414,6 +499,22 @@ class OptionMenuList extends Component {
             )}
           </div>
         )}
+        {navigationButton && (
```

Comment:
This `render` method is getting pretty long. You might create a new method called `renderNavigationMenu` like:

`{navigationButton && renderNavigationButton()}`

or, if you ditch the navigationButton boolean:

`{navigationLabel && renderNavigationButton()}`